Minor manpage polish, meson cleanups and man de-duplication

evelikov opened the pull request:
Random collection of patches, while browsing through https://github.com/kmod-project/kmod/pull/397

First patch is a squash (+ drop of kmod bits) from Ricardo, the other 3 minor man page nitpicks.

The next 4 are meson/autotools yak shaving.

With the last two de-duplicating man sections + an open question how to credit the other 20 individuals in the COPYRIGHT section.

Patches can be picked individually or as a whole. Happy to split into separate PRs - let me know if you have any preferences.

ricardobranco777 committed:
man: drop optional qualifiers for file/modulename

Most tools don't consider the file/module name as optional, so we can
drop the square brackets denoting so.

Mind you, if we want to be extra pedantic - it is not required for
--help, --version et al. Although that should be obvious, as we are
(starting to) simplify yet enhance the documentation.

Signed-off-by: Ricardo Branco <rbranco@suse.de>
[Emil: drop kmod hunk, squash + commit message]
Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>
Reviewed-by: Lucas De Marchi <demarchi@kernel.org>
Link: https://github.com/kmod-project/kmod/pull/pull/400
Signed-off-by: Lucas De Marchi <demarchi@kernel.org>

evelikov committed:
man: remove inconsistent ellipsis

In a number of places we are using ellipsis to indicate plural/multiple
instances. It's not fairly consistent plus we can live without them in
my opinion.

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>
Reviewed-by: Lucas De Marchi <demarchi@kernel.org>
Link: https://github.com/kmod-project/kmod/pull/pull/400
Signed-off-by: Lucas De Marchi <demarchi@kernel.org>

evelikov committed:
rmmod: s/list of modulenames/modulenames/

Drop the "list of" part - we are already (somewhat) consistently using
modulenames across the tree.

While in here, also drop the square brackets - the modulename(s) is not
an optional argument.

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>
Reviewed-by: Lucas De Marchi <demarchi@kernel.org>
Link: https://github.com/kmod-project/kmod/pull/pull/400
Signed-off-by: Lucas De Marchi <demarchi@kernel.org>

evelikov committed:
man: underline the module/file name in modinfo(8)

We do so in all other manual pages, so update the outlier.

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>
Reviewed-by: Lucas De Marchi <demarchi@kernel.org>
Link: https://github.com/kmod-project/kmod/pull/pull/400
Signed-off-by: Lucas De Marchi <demarchi@kernel.org>

evelikov committed:
man: remove manual .so links

These are left overs from the autotools-side of the build system.

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>
Reviewed-by: Lucas De Marchi <demarchi@kernel.org>
Link: https://github.com/kmod-project/kmod/pull/pull/400
Signed-off-by: Lucas De Marchi <demarchi@kernel.org>

evelikov committed:
.gitignore: remove majority of listings

With the move to meson. only out-of-tree builds are allowed. As such, we
can remove ~90% of the contents of our .gitignore file(s).

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>
Reviewed-by: Lucas De Marchi <demarchi@kernel.org>
Link: https://github.com/kmod-project/kmod/pull/pull/400
Signed-off-by: Lucas De Marchi <demarchi@kernel.org>

evelikov committed:
meson: reuse f strings where possible

With meson 0.61 we can use f strings, over the explicit .format().
Multi-line f-strings are introduced with meson 0.63, which we don't
require yet, plus it's not like we have many instances.

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>
Reviewed-by: Lucas De Marchi <demarchi@kernel.org>
Link: https://github.com/kmod-project/kmod/pull/pull/400
Signed-off-by: Lucas De Marchi <demarchi@kernel.org>

evelikov committed:
meson: remove build_by_default instances in man/

With earlier meson versions (pre 0.50) build_by_default and install
didn't have great integration. With newer meson versions, as the former
is undefined the latter will control both.

More importantly, the whole meson.build is included only on
get_option('manpages').

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>
Reviewed-by: Lucas De Marchi <demarchi@kernel.org>
Link: https://github.com/kmod-project/kmod/pull/pull/400
Signed-off-by: Lucas De Marchi <demarchi@kernel.org>

evelikov committed:
scripts/build-scdoc.sh: stop using cat

We don't need the cat + extra pipe - sed can take the input filename as
an argument.

This change is meant to ease de-duplication of the man pages - coming
with a later commit.

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>
Reviewed-by: Lucas De Marchi <demarchi@kernel.org>
Link: https://github.com/kmod-project/kmod/pull/pull/400
Signed-off-by: Lucas De Marchi <demarchi@kernel.org>

evelikov committed:
man: de-duplicate AUTHOR/BUGS sections

Move the sections into a dedicated trailer.scd file, which gets added
during manpage generation.

Note, that this removes the following text from kmod.8, which is
intentional:

*kmod* project was started by Lucas De Marchi as a drop-in replacement to
module-init-tools that was maintained by Jon Masters, adding a library (libkmod)
and additional features.

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>
Reviewed-by: Lucas De Marchi <demarchi@kernel.org>
Link: https://github.com/kmod-project/kmod/pull/pull/400
Signed-off-by: Lucas De Marchi <demarchi@kernel.org>

evelikov committed:
man: remove outdated COPYRIGHT section

While the section is fairly common (11.9k instances of 27.5k man pages
in my Arch install) it is not as ubiquitous as the rest.

Furthermore, the information is severely outdated.

Just remove it all together, instead of trying to devise some clever way
of managing it.

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>
Reviewed-by: Lucas De Marchi <demarchi@kernel.org>
Link: https://github.com/kmod-project/kmod/pull/pull/400
Signed-off-by: Lucas De Marchi <demarchi@kernel.org>

lucasdemarchi commented on testsuite/module-playground/.gitignore:
```diff
@@ -1,11 +0,0 @@
-*.o.d
-
-modules.order
-Module.symvers
```
commenting here because github doesn't allow me to comment in the proper place. The testsuite/rootfs-pristine/test-loaded/sys/module/btusb/holders/.gitignore and testsuite/rootfs-pristine/test-rootfs/testdir/.gitignore were there to guarantee the dir existence in rootfs

evelikov replied:
Good point, I'll swap them for `.gitkeep` which seem more common/appropriate.

lucasdemarchi reviewed (commented):
A few comments, but overall look good. Thanks and sorry for the delay.

evelikov committed:
testsuite: swap empty .gitignore for .gitkeep

Later commit will be removing some no longer needed .gitignore files
and/or contents.

In a few instances, we want the empty folder to be preserved so swap the
empty .gitignore for the more widely used and obvious (IMHO) .gitkeep.

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>
Reviewed-by: Lucas De Marchi <demarchi@kernel.org>
Link: https://github.com/kmod-project/kmod/pull/pull/400
Signed-off-by: Lucas De Marchi <demarchi@kernel.org>

evelikov commented:
- Reinstate coreutils in Alpine CI - w/o it `ln -sfr` seems to choke.

We can drop it at misc later point when their busybox/toybox/etc is updated

evelikov committed:
meson: remove default install:false for static libs

The default for install is false... for static/shared libraries and
executables at least. The documentation for custom_target doesn't
specify.

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>
Reviewed-by: Lucas De Marchi <demarchi@kernel.org>
Link: https://github.com/kmod-project/kmod/pull/pull/400
Signed-off-by: Lucas De Marchi <demarchi@kernel.org>

evelikov committed:
meson: wrap the whole tools section inside an if block

At the moment, we leave random bits out of the `if get_option('tools')`
block (internal/static libkmod, kmod.pc and internal symlinks).

The latter of which, is annotated as build_by_default:true, which means
it always pulls the executable - kmod - and ultimatelly, we build all
tools related artefacts even with -Dtools=false.

Just wrap the whole section and simplify the executable() a bit. Namely:
use install: true and drop build_by_default.

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>
Reviewed-by: Lucas De Marchi <demarchi@kernel.org>
Link: https://github.com/kmod-project/kmod/pull/pull/400
Signed-off-by: Lucas De Marchi <demarchi@kernel.org>

evelikov committed:
meson: use identical ln arguments internally

Update the internal symlinks to also use -r/--relative, just like the
installed ones.

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>
Reviewed-by: Lucas De Marchi <demarchi@kernel.org>
Link: https://github.com/kmod-project/kmod/pull/pull/400
Signed-off-by: Lucas De Marchi <demarchi@kernel.org>

lucasdemarchi commented:
Applied, thanks
